### ISSUE TITLE
BallerinaStreamsV2SelectAggregationTest.testGroupByQuery failing intermittently

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#14792]
```
[ERROR] testGroupByQuery(org.ballerinalang.stdlib.streams.nativeimpl.BallerinaStreamsV2SelectAggregationTest)  Time elapsed: 1.869 s  <<< FAILURE!
java.lang.AssertionError: expected [90] but found [60]
	at org.ballerinalang.stdlib.streams.nativeimpl.BallerinaStreamsV2SelectAggregationTest.testGroupByQuery(BallerinaStreamsV2SelectAggregationTest.java:61)
```

**Comment by riyafa** [comment]
The log file for this failure attached
[stream_log.txt](https://github.com/ballerina-platform/ballerina-lang/files/3022642/stream_log.txt)

**Comment by Manuri** [comment]
Failed again in two PRs with following errors.

1. 
```bash
[ERROR] Tests run: 145, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 302.451 s <<< FAILURE! - in TestSuite
[ERROR] testGroupByQuery(org.ballerinalang.stdlib.streams.nativeimpl.BallerinaStreamsV2SelectAggregationTest)  Time elapsed: 2.214 s  <<< FAILURE!
java.lang.AssertionError: expected [90] but found [30]
	at org.ballerinalang.stdlib.streams.nativeimpl.BallerinaStreamsV2SelectAggregationTest.testGroupByQuery(BallerinaStreamsV2SelectAggregationTest.java:61)

[ERROR] testTimeLengthQuery2(org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest)  Time elapsed: 7.712 s  <<< FAILURE!
java.lang.AssertionError: Expected events are not received expected [6] but found [5]
	at org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest.testTimeLengthQuery2(BallerinaStreamsV2TimeLengthWindowTest.java:69)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   BallerinaStreamsV2TimeLengthWindowTest.testTimeLengthQuery2:69 Expected events are not received expected [6] but found [5]
[ERROR]   BallerinaStreamsV2SelectAggregationTest.testGroupByQuery:61 expected [90] but found [30]
[INFO] 
[ERROR] Tests run: 145, Failures: 2, Errors: 0, Skipped: 0
```

2. 
```bash
[ERROR] testStreamingStatePersistence(org.ballerinalang.stdlib.streams.BallerinaStreamsV2StatePersistenceTest)  Time elapsed: 12.032 s  <<< FAILURE!
java.lang.AssertionError: expected [60] but found [30]
	at org.ballerinalang.stdlib.streams.BallerinaStreamsV2StatePersistenceTest.testStreamingStatePersistence(BallerinaStreamsV2StatePersistenceTest.java:89)

[ERROR] testTimeLengthQuery2(org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest)  Time elapsed: 0.028 s  <<< FAILURE!
org.ballerinalang.util.exceptions.BLangRuntimeException: 
error: {ballerina/task}TaskError {message:"Cannot initialize the Task Listener/Scheduler."}
	at ballerina/task:init(<native>)
	   ballerina/task:__init(listener.bal:30)
	   ballerina/task:__init(scheduler.bal:22)
	   ballerina/streams:__init(scheduler.bal:35)
	   ballerina/streams:__init(windows.bal:1448)
	   ballerina/streams:timeLength(windows.bal:1613)
	   testTimeLengthwindow(streamingv2-time-length-window-test2.bal:78)
	   startTimeLengthwindowTest2(streamingv2-time-length-window-test2.bal:55)
	at org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest.testTimeLengthQuery2(BallerinaStreamsV2TimeLengthWindowTest.java:66)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   BallerinaStreamsV2StatePersistenceTest.testStreamingStatePersistence:89 expected [60] but found [30]
[ERROR]   BallerinaStreamsV2TimeLengthWindowTest.testTimeLengthQuery2:66 Â» BLangRuntime ...
[INFO] 
[ERROR] Tests run: 145, Failures: 2, Errors: 0, Skipped: 0
```

**Comment by MaryamZi** [comment]
The following failures were also observed for different PRs, all since yesterday. The Jenkins build is unstable too.
```
 testSelectQuery(org.ballerinalang.stdlib.streams.inline.BallerinaStreamsV2AggregatorWithGroupByTest)  Time elapsed: 0.044 s  <<< FAILURE!
org.ballerinalang.util.exceptions.BLangRuntimeException: 
error: {ballerina/task}TaskError {message:"Cannot initialize the Task Listener/Scheduler."}
	at ballerina/task:init(<native>)
	   ballerina/task:__init(listener.bal:30)
	   ballerina/task:__init(scheduler.bal:22)
	   ballerina/streams:ballerina/streams.<init>(snapshot.bal:41)
	at org.ballerinalang.stdlib.streams.inline.BallerinaStreamsV2AggregatorWithGroupByTest.testSelectQuery(BallerinaStreamsV2AggregatorWithGroupByTest.java:52)
```

```
testStreamJoinQuery(org.ballerinalang.stdlib.streams.inline.BallerinaStreamsV2JoinTest)  Time elapsed: 0.074 s  <<< FAILURE!
org.ballerinalang.util.exceptions.BLangRuntimeException: 
error: {ballerina/task}TaskError {message:"Cannot initialize the Task Listener/Scheduler."}
	at ballerina/task:init(<native>)
	   ballerina/task:__init(listener.bal:30)
	   ballerina/task:__init(scheduler.bal:22)
	   ballerina/streams:ballerina/streams.<init>(snapshot.bal:41)
	at org.ballerinalang.stdlib.streams.inline.BallerinaStreamsV2JoinTest.testStreamJoinQuery(BallerinaStreamsV2JoinTest.java:52)
```

```
testTimeLengthQuery2(org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest)  Time elapsed: 0.01 s  <<< FAILURE!
org.ballerinalang.util.exceptions.BLangRuntimeException: 
error: {ballerina/task}TaskError {message:"Cannot initialize the Task Listener/Scheduler."}
	at ballerina/task:init(<native>)
	   ballerina/task:__init(listener.bal:30)
	   ballerina/task:__init(scheduler.bal:22)
	   ballerina/streams:__init(scheduler.bal:35)
	   ballerina/streams:__init(windows.bal:1448)
	   ballerina/streams:timeLength(windows.bal:1613)
	   testTimeLengthwindow(streamingv2-time-length-window-test2.bal:78)
	   startTimeLengthwindowTest2(streamingv2-time-length-window-test2.bal:55)
	at org.ballerinalang.stdlib.streams.BallerinaStreamsV2TimeLengthWindowTest.testTimeLengthQuery2(BallerinaStreamsV2TimeLengthWindowTest.java:66)
```

Couldn't it be some other issue?

**Comment by ThisaruGuruge** [comment]
Hi @MaryamZi what you are pointing at is a separate issue. #14848 will fix it.

**Comment by gimantha** [comment]
Fixed with #14701
